### PR TITLE
Redpanda: taking proper care of configs that are arrays

### PR DESCRIPTION
**Pull request by alejandroEsc**
Fixes: https://github.com/redpanda-data/team-kubernetes-internal/issues/14

### CODE REVIEW

**Review by RafalKorepta — CHANGES_REQUESTED** [review]
Could you add test?

**Comment by RafalKorepta** [comment]
I would leverage https://docs.redpanda.com/current/reference/cluster-properties/#kafka_nodelete_topics and create test Pod that tries delete topic.

File: charts/redpanda/templates/tests/test-kafka-nodelete.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          {{- else }}
+          REDPANDA_SASL_MECHANISM=${REDPANDA_SASL_MECHANISM:-{{ .Values.auth.sasl.mechanism | upper }}}
+          {{- end }}
+          export {{ include "rpk-sasl-environment-variables" . }}
+          if [[ -n "$old_setting" ]]; then set -x; fi
+{{- end }}
+          {{- $i := .Values.statefulset.replicas }}
+          {{- $default_topic_replicas := sub (add $i (mod $i 2)) 1 }}
+          # wait for post-upgrade job to update the default_topic_replications value
+          timeout 600 bash -c "until [[ $(rpk cluster config get default_topic_replications) = {{ $default_topic_replicas }} ]]; do sleep 1; done"
```
Comment (RafalKorepta) [inline charts/redpanda/templates/tests/test-kafka-nodelete.yaml:75]:
```suggestion
          timeout 120 bash -c "until [[ $(rpk cluster config get default_topic_replications) = {{ $default_topic_replicas }} ]]; do sleep 1; done"
```

The time out should not be grater than 120 seconds defined in https://github.com/redpanda-data/helm-charts/pull/905/files#diff-3f599d2fdc6489c00bfddc47fc46223cd8fa1dc87c65833bd5f0ecfac9326aeeR51

File: charts/redpanda/templates/tests/test-kafka-nodelete.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  
+          exists=$(rpk topic list | grep my_sample_topic | awk '{print $1}')
```
Comment (RafalKorepta) [inline charts/redpanda/templates/tests/test-kafka-nodelete.yaml:77]: If possible the `helm test` command should be successful even if `kafka_nodelete_topics` is defined but lack of `my_sample_topic`.

Simple check in go temaplte if array has `my_sample_topic` included.

Comment (alejandroEsc) [reply charts/redpanda/templates/tests/test-kafka-nodelete.yaml:77]: ok, ill add a check

File: charts/redpanda/templates/tests/test-kafka-nodelete.yaml [hunk 3]
```diff
@@ -89,12 +90,14 @@ spec:
          
           # now check if we can delete the topic (we should not)
           rpk topic delete my_sample_topic
+
+          {{- if has "my_sample_topic" $noDeleteTopics }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/tests/test-kafka-nodelete.yaml:94]: I was thinking that check would be at the top, so the Pod would not be executed at all.

**Pull request merged**